cloud/ec2_util.go

```diff
@@ -310,6 +316,115 @@ func cacheHostData(ctx context.Context, h *host.Host, instance *types.Instance,
 		}
 	}
 
+	if h.NoExpiration {
+		grip.Error(message.WrapError(setHostPersistentDNSName(ctx, env, h, instance, client), message.Fields{
+			"message":    "could not update host's persistent DNS name",
+			"op":         "upsert",
+			"dashboard":  "evergreen sleep schedule health",
+			"host_id":    h.Id,
+			"started_by": h.StartedBy,
+		}))
+	}
+
+	return nil
+}
+
+// persistentDNSRecordTTLSecs is the number of seconds that a DNS record can be
+// cached. It's intentionally very low so that changes to the DNS name's IP
+// address propagate quickly to users.
+const persistentDNSRecordTTLSecs = 1
+
+// setHostPersistentDNSName sets a host's persistent DNS record with its
+// associated IP address and sets it on the host.
+func setHostPersistentDNSName(ctx context.Context, env evergreen.Environment, h *host.Host, instance *types.Instance, client AWSClient) error {
+	ipAddr := utility.FromStringPtr(instance.PublicIpAddress)
+	if ipAddr == "" {
+		return errors.New("instance did not include an IP address")
+	}
+
+	settings := env.Settings()
+	if settings.Providers.AWS.PersistentDNS.Domain == "" || settings.Providers.AWS.PersistentDNS.HostedZoneID == "" {
+		// TODO (DEVPROD-4040): once all unexpirable hosts have a persistent DNS
+		// name assigned, this should return an error because not having these
+		// settings will prevent persistent DNS names from updating properly.
```

Comment:
Is there a way to assign them when we hit this?

Reply:
We can't because these are admin settings that are meant to be manually assigned once the feature is ready for users to use. Auto-setting isn't something we do because then we have to hard-code production values into the source code + it would auto-enable the feature when we may not be ready to turn it on.